src/main/java/org/valdroz/vscript/BaseNode.java

```diff
@@ -330,7 +330,17 @@ public Variant execute(VariantContainer variantContainer) {
                 result = Variant.fromArray(arrItems);
             }
             break;
-
+            case NT_MF_IF: {
+                List<Variant> parameterItems = Lists.newArrayList();
+                if (params != null) {
```

Comment:
Implemention of `if` function should evaluate first parameter, and then based on the result `true`/`false`, evaluate and respond with result from either second or third parameter. All parameters should never be evaluated.

Reply:
updated based on your comments. thanks